documentation/src/main/asciidoc/userguide/chapters/query/hql/HQL.adoc

```diff
@@ -1351,7 +1351,83 @@ Applications that aim for database portability should avoid using functions in t
 
 Application developers can also supply their own set of functions.
 This would usually represent either custom SQL functions or aliases for snippets of SQL.
-Such function declarations are made by using the `addSqlFunction()` method of `org.hibernate.cfg.Configuration`.
+Such function declarations are made by using the `addSqlFunction()` method of
+the `org.hibernate.boot.MetadataBuilder` or the legacy `org.hibernate.cfg.Configuration`.
+
+Now, let's assume we have the following `apply_vat` PostgreSQL user-defined function:
+
+[[hql-custom-function-postgresql-example]]
+.Custom PostgreSQL function
+====
+[source, JAVA, indent=0]
+----
+include::{sourcedir}/PostgreSQLFunctionWhereClauseTest.java[tags=hql-custom-function-postgresql-example]
+----
+====
+
+Let's consider we have persisted the following entity in our database:
+
+[[hql-custom-function-postgresql-entity-example]]
+.Book entity
+====
+[source, JAVA, indent=0]
+----
+include::{sourcedir}/PostgreSQLFunctionWhereClauseTest.java[tags=hql-custom-function-postgresql-entity-example]
+----
+====
+
+[[hql-non-standard-functions-where-clause]]
+==== Non-standardized functions referenced in the WHERE clause
+
+By default, Hibernate can pass through any user-defined function that's being used in the WHERE clause
+of a JPQL/HQL entity query.
```

Comment:
So why do we have issues like https://hibernate.atlassian.net/browse/HHH-3992 ?

Reply:
Because for a user-defined function, Hibernate expects it to return something. So, if CONTAINS in SQL Server returned a Boolean value, everything would work:

    WHERE CONTAINS( pp.keywords, :keywords ) = TRUE

But CONTAINS in SQL Server should be used like this instead:

    WHERE CONTAINS( pp.keywords, :keywords )

And Hibernate does not support a missing right-hand side.